plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/ServerConfig.java

```diff
@@ -0,0 +1,232 @@
+     * Sets system properties for Arrow memory management and configures thread pool settings.
+     *
+     * @param settings The OpenSearch settings to initialize the server with
+     */
+    @SuppressForbidden(reason = "required for arrow allocator")
+    @SuppressWarnings("removal")
+    public static void init(Settings settings) {
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            System.setProperty("arrow.allocation.manager.type", ARROW_ALLOCATION_MANAGER_TYPE.get(settings));
+            System.setProperty("arrow.enable_null_check_for_get", Boolean.toString(ARROW_ENABLE_NULL_CHECK_FOR_GET.get(settings)));
+            System.setProperty("arrow.enable_unsafe_memory_access", Boolean.toString(ARROW_ENABLE_UNSAFE_MEMORY_ACCESS.get(settings)));
+            System.setProperty("arrow.memory.debug.allocator", Boolean.toString(ARROW_ENABLE_DEBUG_ALLOCATOR.get(settings)));
+            Netty4Configs.init(settings);
+            return null;
+        });
+        enableSsl = ARROW_SSL_ENABLE.get(settings);
+        threadPoolMin = FLIGHT_THREAD_POOL_MIN_SIZE.get(settings);
+        threadPoolMax = FLIGHT_THREAD_POOL_MAX_SIZE.get(settings);
+        keepAlive = FLIGHT_THREAD_POOL_KEEP_ALIVE.get(settings);
+    }
+
+    /**
+     * Checks if SSL/TLS is enabled for the Flight server.
+     *
+     * @return true if SSL is enabled, false otherwise
+     */
+    public static boolean isSslEnabled() {
+        return enableSsl;
+    }
+
+    /**
+     * Gets the thread pool executor builder configured for the Flight server.
+     *
+     * @return The configured ScalingExecutorBuilder instance
+     */
+    public static ScalingExecutorBuilder getServerExecutorBuilder() {
+        return new ScalingExecutorBuilder(FLIGHT_SERVER_THREAD_POOL_NAME, threadPoolMin, threadPoolMax, keepAlive);
+    }
+
+    /**
+     * Gets the thread pool executor builder configured for the Flight server.
+     *
+     * @return The configured ScalingExecutorBuilder instance
+     */
+    public static ScalingExecutorBuilder getClientExecutorBuilder() {
+        return new ScalingExecutorBuilder(FLIGHT_CLIENT_THREAD_POOL_NAME, threadPoolMin, threadPoolMax, keepAlive);
+    }
+
+    /**
+     * Returns a list of all settings managed by this configuration class.
+     *
+     * @return List of Setting instances
+     */
+    public static List<Setting<?>> getSettings() {
+        return new ArrayList<>(
+            Arrays.asList(
+                ARROW_ALLOCATION_MANAGER_TYPE,
+                ARROW_ENABLE_NULL_CHECK_FOR_GET,
+                ARROW_ENABLE_DEBUG_ALLOCATOR,
+                ARROW_ENABLE_UNSAFE_MEMORY_ACCESS,
+                ARROW_SSL_ENABLE
+            )
+        ) {
+            {
+                addAll(Netty4Configs.getSettings());
+            }
+        };
+    }
+
+    static EventLoopGroup createELG(String name, int eventLoopThreads) {
+
+        return Epoll.isAvailable()
```

Comment:
Sorry, missed that: I believe `DefaultThreadFactory` won't work with SM (still, there): you are better off using `OpenSearchExecutors.daemonThreadFactory`

Reply:
changed, thanks